src/main/java/com/eerussianguy/blazemap/api/maps/IClientComponent.java

```diff
@@ -0,0 +1,21 @@
+package com.eerussianguy.blazemap.api.maps;
+
+import com.eerussianguy.blazemap.api.BlazeRegistry;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.Level;
+
+/**
+ * To unify the rendering behaviour of widgets that can take either a MapType or a Layer
+ */
+public interface IClientComponent {
+    public BlazeRegistry.Key<? extends IClientComponent> getID();
```

Comment:
I missed something here.
Instead of declaring this method, this interface should extend `RegistryEntry`, this forces the method signature to be sourced from the same definition.

Reply:
Done